File: ontrack-extension-jira/src/main/java/net/nemerosa/ontrack/extension/jira/model/JIRAIssue.java
package net.nemerosa.ontrack.extension.jira.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.experimental.Wither;
import net.nemerosa.ontrack.extension.issues.model.Issue;

import java.time.LocalDateTime;
import java.util.List;

@Data
@AllArgsConstructor
public class JIRAIssue implements Issue {

    private final String url;
    private final String key;
    private final String summary;
    private final JIRAStatus status;
    private final String assignee;
    private final LocalDateTime updateTime;
    private final List<JIRAField> fields;
    private final List<JIRAVersion> affectedVersions;
    private final List<JIRAVersion> fixVersions;
    private final String issueType;
    @Wither
    private final List<JIRALink> links;

}

File: ontrack-extension-jira/src/main/java/net/nemerosa/ontrack/extension/jira/model/JIRAIssue.kt
package net.nemerosa.ontrack.extension.jira.model

import net.nemerosa.ontrack.extension.issues.model.Issue
import java.time.LocalDateTime

class JIRAIssue(
    val id: String,
    override val url: String,
    override val key: String,
    override val summary: String,
    override val status: JIRAStatus,
    val assignee: String,
    override val updateTime: LocalDateTime,
    val fields: List<JIRAField>,
    val affectedVersions: List<JIRAVersion>,
    val fixVersions: List<JIRAVersion>,
    val issueType: String,
    val links: List<JIRALink>
) : Issue {
    fun withLinks(links: List<JIRALink>) = JIRAIssue(
        id = id,
        url = url,
        key = key,
        summary = summary,
        status = status,
        assignee = assignee,
        updateTime = updateTime,
        fields = fields,
        affectedVersions = affectedVersions,
        fixVersions = fixVersions,
        issueType = issueType,
        links = links
    )
}
